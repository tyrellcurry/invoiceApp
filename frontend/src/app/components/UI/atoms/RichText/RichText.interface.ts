export interface IRichTextProps {
  className?: string;
  value: React.ReactNode;
}
