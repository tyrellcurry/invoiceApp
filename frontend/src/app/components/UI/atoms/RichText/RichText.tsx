/**
 * @name RichText
 * @author Tyrell Curry <tyrellcurryio@gmail.com>
 *
 * Used for rendering HTML as Rich Text
 *
 * @param className - element's custom classes
 * @param(required) value - HTML or string to render
 *
 * @returns {JSX.Element}
 */

import { JSX } from 'react';
import { IRichTextProps } from './RichText.interface';
import classNames from 'classnames';

const RichText = (props: IRichTextProps): JSX.Element => {
  const { className: propsClassName, value, ...rest } = props;
  const className = classNames('text text--rich', propsClassName);
  return (
    <div {...rest} {...{ className }}>
      {value}
    </div>
  );
};

export default RichText;
