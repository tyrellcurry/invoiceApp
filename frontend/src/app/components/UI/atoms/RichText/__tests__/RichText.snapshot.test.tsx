import React from 'react';
import { render } from '@testing-library/react';
import RichText from '@/app/components/UI/atoms/RichText/RichText';

describe('RichText Component', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<RichText value="Hello, World!" />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with HTML passed as a value', () => {
    const { container } = render(<RichText value={<p>Hello World</p>} />);
    expect(container).toMatchSnapshot();
  });
});
