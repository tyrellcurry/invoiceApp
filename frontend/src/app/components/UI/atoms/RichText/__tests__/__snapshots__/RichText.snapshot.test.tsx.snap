// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText Component renders correctly with HTML passed as a value 1`] = `
<div>
  <div
    class="text--rich"
  >
    <p>
      Hello World
    </p>
  </div>
</div>
`;

exports[`RichText Component renders correctly with default props 1`] = `
<div>
  <div
    class="text--rich"
  />
</div>
`;
